indexers/vector/ZarrIndexer/__init__.py

```diff
@@ -0,0 +1,63 @@
+from os import path
+from typing import List, Union, Optional
+
+import zarr
+import numpy as np
+
+from jina.executors.indexers.vector import NumpyIndexer
+from jina.helper import cached_property
+
+
+class ZarrIndexer(NumpyIndexer):
+    """Indexing based on Zarr arrays
+    
+    For more information about Zarr, please check
+    https://zarr.readthedocs.io/en/stable/index.html
+    
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        
+    def get_add_handler(self) -> zarr.hierarchy.Group:
+        """Open an existing zarr group file for adding new vectors
+
+        :return: a zarr group file
+        """
+        return zarr.open(store=self.index_abspath, mode='a')
+
+    def get_create_handler(self) -> zarr.hierarchy.Group:
+        """Create a zarr group file for adding new vectors
+
+        :return: a zarr group file
+        """
+        return zarr.open(store=self.index_abspath, mode='w')
+    
+    def add(self, keys: 'np.ndarray', vectors: 'np.ndarray', *args, **kwargs) -> None:
+        self._validate_key_vector_shapes(keys, vectors)
+        if 'default' in self.write_handler.array_keys():
```

Comment:
what does this do?

Reply:
We are creating a group first, since we cannot know the array size, and without knowing the array size. we cannot create a `zarr.core.Array`. And inside that group, we are creating an array called `default`. This is done in the `add` method using `write_handler`. Hence in `query_handler`, we are just verifying that.

Reply:
`query_handler` is normally used without the `write_handler` initialized I think, at least I think is how most of the times it behaves, since we are splitting index and querying time

Reply:
Looking at other Indexers, I have removed this check. Directly reading the array now,